src/main/java/subway/application/SectionService.java

```diff
@@ -27,32 +28,56 @@ public SectionService(SectionDao sectionDao, LineDao lineDao, StationDao station
-        Line line = lineDao.findById(lineId);
-        Section section = new Section(
-            upStation,
-            downStation,
-            line,
-            sectionRegistRequest.getDistance()
-        );
-
+    public void registerSection(SectionRegisterRequest sectionRegisterRequest, Long lineId) {
+        Section section = makeSection(sectionRegisterRequest, lineId);
         Sections sections = sectionDao.findAllByLineId(lineId);
-        SectionRegistVo result = sections.registSection(section);
 
-        sectionDao.insert(result.getAddSection());
+        sections.validRegisterSection(section);
 
-        if (result.getUpdateSection().isPresent()) {
-            sectionDao.update(result.getUpdateSection().get());
-        }
+        sectionDao.insert(section);
+        sections.makeUpdateSection(section).ifPresent(sectionDao::update);
+    }
+
+    private Section makeSection(SectionRegisterRequest sectionRegisterRequest, Long lineId) {
+        Station upStation = stationDao.findById(sectionRegisterRequest.getUpStationId());
+        Station downStation = stationDao.findById(sectionRegisterRequest.getDownStationId());
+        Line line = lineDao.findById(lineId);
+        return new Section(
+                upStation,
+                downStation,
+                line,
+                sectionRegisterRequest.getDistance()
+        );
     }
 
     @Transactional
     public void deleteSection(Long stationId, Long lineId) {
         Sections sections = sectionDao.findAllByLineId(lineId);
-        sections.canDeleteStation(stationId);
+        Station deleteStation = stationDao.findById(stationId);
+        sections.validDeleteStation(deleteStation);
+
+        Optional<Section> sectionByUpStation = sections.findSectionByUpStation(deleteStation);
+        Optional<Section> sectionByDownStation = sections.findSectionByDownStation(deleteStation);
+
+        deleteSectionsContainingDeleteStation(sectionByUpStation, sectionByDownStation);
+
+        makeCombineSection(sectionByUpStation, sectionByDownStation);
+    }
+
+    private void deleteSectionsContainingDeleteStation(
+            Optional<Section> sectionByUpStation,
+            Optional<Section> sectionByDownStation
+    ) {
+        sectionByUpStation.ifPresent(section -> sectionDao.deleteById(section.getId()));
+        sectionByDownStation.ifPresent(section -> sectionDao.deleteById(section.getId()));
+    }
 
-        sectionDao.deleteByDownStationIdAndLineId(stationId, lineId);
+    private void makeCombineSection(Optional<Section> sectionByUpStation, Optional<Section> sectionByDownStation) {
+        if (sectionByDownStation.isPresent() && sectionByUpStation.isPresent()) {
+            Section newSectionUpStation = sectionByDownStation.get();
+            Section newSectionDownStation = sectionByUpStation.get();
+            Section newSection = newSectionUpStation.combineSection(newSectionDownStation);
+            sectionDao.insert(newSection);
+        }
```

Comment:
```suggestion
        sectionByUpStation.ifPresent( 
            section -> {
                Section newSection = section.combineSection(sectionByDownStation.orElseThrow());
                sectionDao.insert(newSection);
            }
        );
    }
```

optional을 더 잘 활용해보면 위와 같이 구현해 보아도 좋을 것 같아요 ! 
예외 메시지나 네이밍을 조금 더 다듬어야하겠지만요 ㅎㅎ (개인적인 의견입니다)

Reply:
이 부분에 대해 더 좋은 방법이 있나 고민 많이 했었는데 좋은 방법 알려주셔서 감사합니다!